lib/spacesuit/proxy_handler.ex

```diff
@@ -23,6 +24,7 @@ defmodule Spacesuit.ProxyHandler do
     # Make the proxy request
     req = handle_request(req, ups_url, ups_headers, method)
 
+    NewRelic.stop_transaction()
```

Comment:
What happens if something fails before this. Do we have a Elixir equivalent of `defer` ???

Reply:
No. In Elixir exceptions are not used at all but they can happen due to some things like an incorrect pattern match. Reading the code I don't see any point that could potentially generate an exception. That being said I just added a `try/after` statement to handle that case.